tests/integration/testsuites/gateway/scenario.go

```diff
@@ -0,0 +1,279 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"github.com/cucumber/godog"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/helpers"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/manifestprocessor"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/resource"
+	"github.com/kyma-project/api-gateway/tests/integration/pkg/testcontext"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"log"
+	"path"
+	"time"
+)
+
+const manifestsPath = "testsuites/gateway/manifests/"
+
+type scenario struct {
+	testID          string
+	namespace       string
+	k8sClient       dynamic.Interface
+	resourceManager *resource.Manager
+	config          testcontext.Config
+}
+
+func initScenario(ctx *godog.ScenarioContext, ts *testsuite) {
+	scenario, err := createScenario(ts)
+
+	if err != nil {
+		log.Fatalf("could not initialize custom domain endpoint err=%s", err)
+	}
+
+	ctx.Step(`^there is an APIGateway operator in "([^"]*)" state$`, scenario.thereIsAnAPIGatewayOperator)
```

Comment:
It's the CR, not the operator. We should also adopt the function name in this scenario
```suggestion
	ctx.Step(`^there is an APIGateway CR in "([^"]*)" state$`, scenario.thereIsAnAPIGatewayOperator)
```